.github/workflows/ci.yml

```diff
@@ -48,10 +49,15 @@ jobs:
             --disable-maintainer-mode \
             --disable-valgrind \
             --with-oniguruma=builtin \
-            ${{ matrix.configure_flag }}
+            --host=${{ matrix.arch }}-linux-gnu \
```

Comment:
I started looking into doing cross builds yesterday but ended up reading old autoconf documentation about host, target and build and got irritated 😄 what a mess. Also a bit confused why jq's configure script only has `--build` and `--host`... and `--host` is what i would call "target"? :) it's generated by autoconf 2.71 which is quite modern.

Anyways thanks doing this!

Reply:
@wader https://github.com/jqlang/jq/wiki/Cross-compilation

Reply:
I added cross-compilation for Windows as well. However, I noticed a few things:

* Tests with the `mingw-w64-x86_64` build [failed](https://github.com/owenthereal/jq/actions/runs/5498251723/jobs/10019578971) for segmentation fault:  (Maybe the same issue as the [clang issue](https://discord.com/channels/1108476965947723846/1119871041997652038/1122924762075758723)). I fallback to using msys gcc for x86_64 build
* For Windows i386 build, I needed to add the `__MINGW32__` guard to include `include <pthread.h>` instead of the adding `pthread_*` functions defined in `jv.c` & `jv_thread.h` ([Ref](https://github.com/owenthereal/jq/actions/runs/5496335434/jobs/10016293755#step:4:1237)).